extensions/wikia/SpecialDiscussions/SpecialDiscussionsController.class.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+use Swagger\Client\ApiException;
+use Swagger\Client\Discussion\Api\SitesApi;
+use Wikia\DependencyInjection\Injector;
+use Wikia\Service\Swagger\ApiProvider;
+
+/**
+ * Discussion user log page
+ */
+class SpecialDiscussionsController extends WikiaSpecialPageController {
+
+	const HTTP_STATUS_OK = 200;
+	const DISCUSSIONS_ACTION = 'specialdiscussions';
+	const SERVICE_NAME = 'discussion';
+	const TIMEOUT = 5;
+	const EDIT_TOKEN = 'editToken';
+
+	const DEFAULT_TEMPLATE_ENGINE = \WikiaResponse::TEMPLATE_ENGINE_MUSTACHE;
+
+	private $logger;
+	private $sitesApi;
+	private $siteId;
+	private $siteName;
+
+	public function __construct() {
+		parent::__construct( 'Discussions', '', false );
+		$this->sitesApi = $this->getDiscussionSitesApi();
+		$this->siteId = F::app()->wg->CityId;
+		$this->siteName = WikiFactory::getVarValueByName( 'wgSitename', $this->siteId );
```

Comment:
The value of `$this->siteId` just came from `F::app()->wg->CityId` so is there any reason why we can't get siteName from `F::app()->wg->Sitename`?  Calling WikiFactory here is going to make a DB call.